File: leetcode-arry/189.旋转数组/python/rotate-array.py
class Solution:
    def rotate(self, nums: List[int], k: int) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        new_nums = nums
        nums= []
        for i in range(len(new_nums)-k,len(new_nums)):
            nums.append(new_nums[i])
        for i in range(0,len(new_nums)-k):
            nums.append(new_nums[i])

File: leetcode-arry/189.旋转数组/java/Rotate Array.java
class Solution {
    public void rotate(int[] nums, int k) {
        int[] result = new int[nums.length];
        for(int i = 0;i < nums.length;i++) {
            result[obtainIndex(i,k,nums.length)] = nums[i];
        }
        System.arraycopy(result,0,nums,0,nums.length);
    }

    private static int obtainIndex(int current,int offset,int length) {
        int v = current + offset;
        return v < length ? v : xx(v,length);
    }

    private static int xx(int offset,int length) {
        int v2 = offset - length;
        return v2 < length ? v2 : xx(v2,length);
    }
}

class Solution2 {
    // 1. 取模
    public void rotate(int[] nums, int k) {
        int[] result = new int[nums.length];
        for(int i = 0;i < nums.length;i++) {
            result[(k + i) % nums.length] = nums[i];
        }
        System.arraycopy(result,0,nums,0,nums.length);
    }
}